apps/system/js/audio_channel_controller.js

```diff
@@ -0,0 +1,211 @@
+/* global BaseUI */
+'use strict';
+
+(function(exports) {
+  var FADE_IN_VOLUME = 1;
+  var FADE_OUT_VOLUME = 0.2;
+  var VIBRATION_DURATION = 1000;
+
+  /**
+   * AudioChannelController controls the audio channel to
+   * play or pause.
+   */
+  var AudioChannelController = function(app, channel) {
+    this.app = app;
+    this.name = channel.name;
+    this._channel = channel;
+    this._states = {
+      active: false,
+      playing: false,
+      fadingOut: false,
+      vibrating: false
+    };
+    this._policy = {};
+    this._generateID();
+    channel.addEventListener('activestatechanged', this);
+    app.element.addEventListener('_terminating', this);
```

Comment:
nit: We don't need this - AppWindow will call this.destroy() before it remove the DOM element.
We could implement AudioChannelController#destroy to deliver this event.